server/parameters/ConfigParser.java

```diff
@@ -0,0 +1,472 @@
+                        assert Reasoner.type.equals(type);
+                        return new Config.Log.Debugger.Reasoner(
+                                output.parse(yaml.asMap(), path), enable.parse(yaml.asMap(), path));
+                    } else throw TypeDBException.of(CONFIG_SECTION_MUST_BE_MAP, path);
+                }
+
+                @Override
+                public List<HelpEntry> helpEntries(String path) {
+                    return list(typeParser.helpEntry(path), output.helpEntry(path), enable.helpEntry(path));
+                }
+            }
+        }
+    }
+
+    private static class VaticleFactory extends Compound<Config.VaticleFactory> {
+
+        private static final String name = "vaticle-factory";
+        private static final String description = "Configure Vaticle Factory connection.";
+
+        private static final Predefined<Boolean> enable =
+                predefined("enable", "Enable Vaticle Factory tracing.", BOOLEAN);
+        private static final Predefined<String> uri =
+                predefined("uri", "URI of Vaticle Factory server.", STRING);
+        private static final Predefined<String> username =
+                predefined("username", "Username for Vaticle Factory server.", STRING);
+        private static final Predefined<String> token =
+                predefined("token", "Authentication token for Vaticle Factory server.", STRING);
+        private static final Set<Predefined<?>> parsers = set(enable, uri, username, token);
+
+        @Override
+        public Config.VaticleFactory parse(Yaml yaml, String path) {
+            if (yaml.isMap()) {
+                boolean trace = enable.parse(yaml.asMap(), path);
+                validatedRecognisedParsers(parsers, yaml.asMap().keys(), path);
+                if (trace) {
+                    return new Config.VaticleFactory(true, uri.parse(yaml.asMap(), path),
+                            username.parse(yaml.asMap(), path), token.parse(yaml.asMap(), path));
+                } else {
+                    return new Config.VaticleFactory(false, null, null, null);
+                }
+            } else throw TypeDBException.of(CONFIG_SECTION_MUST_BE_MAP, path);
+        }
+
+        @Override
+        public List<HelpEntry> helpEntries(String path) {
+            return list(enable.helpEntry(path), uri.helpEntry(path), username.helpEntry(path), token.helpEntry(path));
+        }
+    }
+
+    private static void validatedRecognisedParsers(Set<Predefined<?>> parsers, Set<String> keys, String path) {
+        Set<String> unrecognisedKeys = new HashSet<>(keys);
+        parsers.forEach(parser -> unrecognisedKeys.remove(parser.key()));
+        if (!unrecognisedKeys.isEmpty()) {
+            Set<String> childPaths = iterate(unrecognisedKeys).map(key -> YamlParser.concatenate(path, key)).toSet();
+            throw TypeDBException.of(UNRECOGNISED_CONFIGURATION_OPTIONS, childPaths);
+        }
+    }
+
+}
```

Comment:
class looks good! Splitting out data structure and parser was definitely the right call :)